src/main/java/org/fenixedu/academic/ui/struts/action/manager/ManageAssociatedObjects.java

```diff
@@ -634,7 +640,7 @@ public ActionForward editDepartment(ActionMapping mapping, ActionForm form, Http
             d.setName(bean.getName());
             d.setRealName(bean.getRealName());
             d.setRealNameEn(bean.getRealNameEn());
-            d.setCompetenceCourseMembersGroup(User.findByUsername(bean.getUsername()).groupOf());
+            d.setCompetenceCourseMembersGroup(Group.parse(bean.getUsername()));
```

Comment:
bean ``getUsername`` should be changed to something more semantic like ``membersGroupExpression``

Reply:
Since getUsername method is being used in "Academic Office creation" to link a single user, the bean should just be extended with a membersGroupExpression attribute.